src/utilities/trezorTransformTransaction.js

```diff
@@ -0,0 +1,171 @@
+// https://github.com/trezor/connect/blob/develop/src/js/plugins/stellar/plugin.js
+import StellarSdk from "stellar-sdk";
+import BigNumber from "bignumber.js";
+/**
+ * Transforms StellarSdk.Signer to TrezorConnect.StellarTransaction.Signer
+ */
+const transformSigner = (signer) => {
+  let type = 0;
+  let key;
+  const { weight } = signer;
+  if (typeof signer.ed25519PublicKey === "string") {
+    const keyPair = StellarSdk.Keypair.fromPublicKey(signer.ed25519PublicKey);
+    key = keyPair.rawPublicKey().toString("hex");
+  }
+  if (signer.preAuthTx instanceof Buffer) {
+    type = 1;
+    key = signer.preAuthTx.toString("hex");
+  }
+  if (signer.sha256Hash instanceof Buffer) {
+    type = 2;
+    key = signer.sha256Hash.toString("hex");
+  }
+  return {
+    type,
+    key,
+    weight,
+  };
+};
+/**
+ * Transforms StellarSdk.Asset to TrezorConnect.StellarTransaction.Asset
+ */
+const transformAsset = (asset) => {
+  if (asset.isNative()) {
+    return {
+      type: 0,
+      code: asset.getCode(),
+    };
+  }
+  return {
+    type: asset.getAssetType() === "credit_alphanum4" ? 1 : 2,
+    code: asset.getCode(),
+    issuer: asset.getIssuer(),
+  };
+};
+/**
+ * Transforms amount from decimals (lumens) to integer (stroop)
+ */
+const transformAmount = (amount) =>
+  new BigNumber(amount).times(10000000).toString();
+/**
+ * Transforms StellarSdk.Operation.type to TrezorConnect.StellarTransaction
+ * Operation.type
+ */
+const transformType = (type) => {
+  switch (type) {
+    case "pathPaymentStrictReceive":
+      // case 'pathPaymentStrictSend':
+      return "pathPayment";
+    case "createPassiveSellOffer":
+      return "createPassiveOffer";
+    case "manageSellOffer":
+      // case 'manageBuyOffer':
+      return "manageOffer";

```

Comment:
out of scope for this PR, but we should implement an enum here during the Lab refactor

Reply:
We could refactor this but it might be better to keep it as close to the original created by Trezor as possible. Just in case there are updates. Aside from added types and some minor adjustments, it's pretty close to the original.